File: daily/Sliding Window/424-Longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        n = len(s)
        left = 0
        res = -float('inf')
        seen = defaultdict(int)
        max_frequency = 0
        for right in range(n):
            seen[s[right]] += 1
            max_frequency = max(max_frequency, seen[s[right]])
            while (right - left + 1) - max_frequency > k:
                seen[s[left]] -= 1
                left += 1
            res = max(res, right - left + 1)


        return res if res != -float('inf') else 0

File: daily/Sliding Window/424-Longest-repeating-character-replacement.go
package daily

func characterReplacement(s string, k int) int {
	n := len(s)
	left := 0
	res := 0
	seen := make(map[rune]int)
	maxFrequency := 0
	for right := 0; right < n; right++ {
		seen[rune(s[right])]++
		maxFrequency = max(maxFrequency, seen[rune(s[right])])
		for (right-left+1)-maxFrequency > k {
			seen[rune(s[left])]--
			left++
		}
		res = max(res, right-left+1)
	}
	return res
}
